documentation/API_documentation/Carrier_Billing_Checkout_API.md

```diff
@@ -0,0 +1,241 @@
+ <br>**Response**<br> **201: Purchase created**<br> **Response body**: A complete purchase representation is provided. <br> **payable**: same payable as in request.<br> **id**: unique purchase identifier.<br> **status**: purchase status. <br> **400:** **Invalid Argument.**<br> **401:** **Unauthorized.** <br> **403:** **Permission Denied.** <br> **403:** **Purchase Amount Overpassed.**<br> **500:** **Server Error.**<br> **503:** **Service unavailable.**<br> **504:** **Request timeout exceeded.** |
+<br>
+
+
+| **Retrieve Purchase Information** |
+| --------------------------------------- |
+| **HTTP Request**<br> GET \<base-url>/carrier-billing-checkout/v0/purchases/{purchase_id}<br>**Query Parameters**<br> No query parameters are defined.<br>**Path Parameters**<br> `purchase_id` MUST be present in the request, to identify the purchase process involved. <br>**Request Body Parameters**<br> No request body parameters are defined.<br>
+ <br>**Response**<br> **200: Contains information about a given purchase**<br> **Response body**: A complete purchase representation is provided. <br> **payable**: payable information related to the purchase. Same model as per createPurchase operation.<br> **id**: unique purchase identifier. Same value as in request path parameter.<br> **status**: purchase status. <br> **401:** **Unauthorized.** <br> **403:** **Permission Denied.** <br> **404:** **Not Found.** <br> **500:** **Server Error.**<br> **503:** **Service unavailable.** <br> **504:** **Request timeout exceeded.** |
+<br>
+
+
+| **Create a new Payment** |
+| --------------------------------------- |
+| **HTTP Request**<br> POST \<base-url>/carrier-billing-checkout/v0/purchases/{purchase_id}/payment/prepare<br>**Query Parameters**<br> No query parameters are defined.<br>**Path Parameters**<br> `purchase_id` MUST be present in the request, to identify the purchase process involved in the payment action. <br>**Request Body Parameters**<br> No request body parameters are defined.<br>
+ <br>**Response**<br> **200: Payment Triggered**<br> **Response body**: Information of triggered payment. <br> **id**: unique purchase identifier. Same value as in request path parameter.<br> **status**: purchase status.<br> **payment_otp**: provides the necessary information in order to validate an OTP, including the `validation_endpoint`. OTP code is sent offline to the user (e.g. via SMS), so end-user can input it in the endpoint provided. <br> **401:** **Unauthorized.** <br> **403:** **Permission Denied.** <br> **403:** **Accumulated Purchase Amount Overpassed.** <br> **404:** **Not Found.** <br> **409:** **Already Exists.** <br> **500:** **Server Error.**<br> **503:** **Service unavailable.** <br> **504:** **Request timeout exceeded.** |
```

Comment:
To be sure to understand the mechanism. The server (carrier billing provider) sends a SMS to the enduser with a code (and a message). App request a code to the user. App send the code to the server (carrier billing provider)  to the validation endpoint.

If correct it probably worth it to shortly explain it.
This is probably not the only way to get end user approval right ? we can indicate this as one way to do it.

Reply:
Yes, i am adding explanation. Not unique mechanism. Thanks for the comments!!